### ISSUE TITLE
[eventbus] Support Kafka clustering

### ISSUE THREAD

**Issue by smallfish-root** [looplab/eventhorizon#339]
Hi, When this project can support kafka cluster?

**Comment by smallfish-root** [comment]
![image](https://user-images.githubusercontent.com/75240003/134884481-bea60919-7205-48e5-abfe-217400172183.png)
by kafka

**Comment by smallfish-root** [comment]
![image](https://user-images.githubusercontent.com/75240003/134892434-84f3631c-3df7-4736-960f-8b7ba77dc79b.png)
if don't read old msg,maybe it will lose data not to be handled,so whether open this param for user?

**Comment by maxekman** [comment]
Definitely good suggestion with clustering.

Regarding LastOffset, it means last non-committed message, at least as far as I know. You definitely don't want to read from the beginning, and not from the end of the log either. What are your thoughts on this?

**Comment by smallfish-root** [comment]
> Definitely good suggestion with clustering.
> 
> Regarding LastOffset, it means last non-committed message, at least as far as I know. You definitely don't want to read from the beginning, and not from the end of the log either. What are your thoughts on this?

If the consumer crash, it should consume all the uncommited msg, after it runs again.
By the way,this cqrs can use without DDD? Any examples?

**Comment by JohnRoesler** [comment]
> When this project can support kafka cluster?

I'm using it currently with clustered kafka broker with no issue. You only need to supply one kafka address to talk to the cluster.

> StartOffset

```golang
// StartOffset determines from whence the consumer group should begin
// consuming when it finds a partition without a committed offset.  If
// non-zero, it must be set to one of FirstOffset or LastOffset.
//
// Default: FirstOffset
//
// Only used when GroupID is set
StartOffset int64
```

If I am reading the documentation right, the line "when it finds a partition without a committed offset" suggests this is handling the case where the consumer group (which handles committing offsets it has consumed to a kafka internal topic to avoid other consumers in the group also consuming them) has restarted or just started up and it gets placed on a partition with no committed offsets, meaning no-one in the group has consumed from this partition yet, then I would think you would want to go with segmentio's default value which is `FirstOffset` as we do want to consume all the offsets.

I am not a kafka expert and could be wrong, but that's my take on the documentation. I tried to find something that confirmed/denied that, but couldn't. If I find the time, I would like to test it out and see the results of each value.

**Comment by maxekman** [comment]
Sounds reasonable to use StartOffset. Can it be set as an option? Would that be a good solution? I’ll happily accept a PR adding an option to set it (to not break existing behavior). Thanks

**Comment by maxekman** [comment]
I believe this is fixed in #387.
